### PR TITLE
fix: Error dialog: due to "show timer" on resume 

### LINKED ISSUE #15317: [Bug]CalledFromWrongThreadException
###### Reproduction Steps

1. On the review screen
2. Press the home button, enter the desktop
3. Back to ankidroid

###### Expected Result

Nothing will happen.

###### Actual Result
A warning dialog box:
android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.

###### Debug info
AnkiDroid Version = 2.17alpha16 (37daacf8e6a9642b85135ce3c00fa44cb58ea061)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 13 (SDK 33)

ProductFlavor = full

Manufacturer = Xiaomi

Model = 22041216C

Hardware = mt6895

Webview User Agent = Mozilla/5.0 (Linux; Android 13; 22041216C Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/104.0.5112.97 Mobile Safari/537.36

ACRA UUID = 13b991d9-a0fd-4ffc-8f8b-0b5c080f71cf

Crash Reports Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
"Show Answer Timer" must be enabled

* background + foreground AnkiDroid
* Error dialog appears

## Fixes
* Fixes #15317 

## Approach
* make `resume() ` a suspend function

## How Has This Been Tested?
Manually + Added `@MainThread`

## Learning (optional, can help others)
`Chronometer.base` must be called on main thread

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/AnswerTimer.kt [hunk 1]
```diff
@@ -119,7 +120,10 @@ class AnswerTimer(private val cardTimer: Chronometer) {
         cardTimer.base = elapsedRealTime - currentCard.timeTaken(col)
         // Don't start the timer if we have already reached the time limit or it will tick over
         if (elapsedRealTime - cardTimer.base < currentCard.timeLimit(col)) {
+            Timber.v("restarting timer")
             cardTimer.start()
+        } else {
+            Timber.v("not restarting timer - at time limit")
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/AnswerTimer.kt]: this commit is in another PR isn't it? in the queue now

**Review by mikehardy — APPROVED** [review]
seems fine but I think there's a duplicated commit as noted

**Comment by david-allison** [comment]
> seems fine but I think there's a duplicated commit as noted

I've removed that from the PR in the merge queue, it should be here.

This occurred due to me committing to my branch which tracks `main`: `develop`

I'll see what I can do with my branching strategy to fix this

**Review by lukstbit — APPROVED** [review]
LGTM!

**Comment by mikehardy** [comment]
Huh. That flake looks new (as opposed to first flake on our typical custom-scheduler-javascript-androidTest one)

```
ReviewerTest > testMultipleCards FAILED
    java.lang.RuntimeException: Can't create handler inside thread Thread[#36,DefaultDispatcher-worker-3 @coroutine#3131,5,SDK 33] that has not called Looper.prepare()
        at android.os.Handler.__constructor__(Handler.java:227)
        at android.os.Handler.<init>(Handler.java)
        at android.app.Dialog.__constructor__(Dialog.java:138)
        at android.app.Dialog.<init>(Dialog.java)
        at androidx.activity.ComponentDialog.<init>(ComponentDialog.kt:42)
        at androidx.appcompat.app.AppCompatDialog.<init>(AppCompatDialog.java:58)
        at androidx.appcompat.app.AlertDialog.<init>(AlertDialog.java:98)
        at androidx.appcompat.app.AlertDialog$Builder.create(AlertDialog.java:982)
        at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:1007)
        at com.ichi2.anki.CoroutineHelpersKt.showError(CoroutineHelpers.kt:468)
        at com.ichi2.anki.CoroutineHelpersKt.showError$default(CoroutineHelpers.kt:215)
        at com.ichi2.anki.CoroutineHelpersKt.runCatchingTask(CoroutineHelpers.kt:[154](https://github.com/ankidroid/Anki-Android/actions/runs/7669499525/job/20903558130#step:8:155))
        at com.ichi2.anki.CoroutineHelpersKt$runCatchingTask$1.invokeSuspend(CoroutineHelpers.kt)
```

**Comment by mikehardy** [comment]
Super annoying

```
ReviewerTest > testLrnQueueAfterUndo FAILED
    java.lang.AssertionError: 
    Expected: "2"
         but: was "1"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.ichi2.anki.RobolectricTest.equalFirstField(RobolectricTest.kt:402)
        at com.ichi2.anki.ReviewerTest$testLrnQueueAfterUndo$1.invokeSuspend(ReviewerTest.kt:170)
        at com.ichi2.anki.ReviewerTest$testLrnQueueAfterUndo$1.invoke(ReviewerTest.kt)
        at com.ichi2.anki.ReviewerTest$testLrnQueueAfterUndo$1.invoke(ReviewerTest.kt)
        at com.ichi2.testutils.TestClass$runTest$1$1.invokeSuspend(TestClass.kt:212)
        at com.ichi2.testutils.TestClass$runTest$1$1.invoke(TestClass.kt)
        at com.ichi2.testutils.TestClass$runTest$1$1.invoke(TestClass.kt)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:314)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:28)
        at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:103)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:320)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:13)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:308)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:[166](https://github.com/ankidroid/Anki-Android/actions/runs/7669624161/job/20903972075#step:8:167))
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at com.ichi2.testutils.TestClass.runTest(TestClass.kt:210)
        at com.ichi2.testutils.TestClass.runTest$default(TestClass.kt:201)
        at com.ichi2.anki.ReviewerTest.testLrnQueueAfterUndo(ReviewerTest.kt:150)```

*and*

```
ReviewerTest > testMultipleCards FAILED
    java.lang.AssertionError: Unexpected card ord
    Expected: <2>
         but: was <1>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.ichi2.anki.ReviewerTest.assertCurrentOrdIs(ReviewerTest.kt:325)
        at com.ichi2.anki.ReviewerTest.answerCardOrdinalAsGood(ReviewerTest.kt:314)
        at com.ichi2.anki.ReviewerTest.access$answerCardOrdinalAsGood(ReviewerTest.kt:51)
        at com.ichi2.anki.ReviewerTest$testMultipleCards$1.invokeSuspend(ReviewerTest.kt:137)
        at com.ichi2.anki.ReviewerTest$testMultipleCards$1.invoke(ReviewerTest.kt)
        at com.ichi2.anki.ReviewerTest$testMultipleCards$1.invoke(ReviewerTest.kt)
        at com.ichi2.testutils.TestClass$runTest$1$1.invokeSuspend(TestClass.kt:212)
        at com.ichi2.testutils.TestClass$runTest$1$1.invoke(TestClass.kt)
        at com.ichi2.testutils.TestClass$runTest$1$1.invoke(TestClass.kt)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:314)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:28)
        at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:103)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:320)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:13)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:308)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:166)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at com.ichi2.testutils.TestClass.runTest(TestClass.kt:[210](https://github.com/ankidroid/Anki-Android/actions/runs/7669624161/job/20903972075#step:8:211))
        at com.ichi2.testutils.TestClass.runTest$default(TestClass.kt:201)
        at com.ichi2.anki.ReviewerTest.testMultipleCards(ReviewerTest.kt:121)
```

**Comment by mikehardy** [comment]
Each of the various tests had all passed at some point so I just merged the thing.

**Comment by david-allison** [comment]
I believe the flake may be due to being an hour away from the rollover timer

**Pull request merged**